pkg/collect/utilisation.go

```diff
@@ -0,0 +1,363 @@
+	memMatrix, ok := memoryResult.(model.Matrix)
+	if !ok {
+		return nil, errors.New("Invalid query result, expected matrix")
+	}
+
+	// Process query result
+	cpuMatrix, ok := cpuResult.(model.Matrix)
+	if !ok {
+		return nil, errors.New("Invalid query result, expected matrix")
+	}
+	metricsByNamespace := make(map[string]map[string][]model.SamplePair, 0)
+	for _, entry := range memMatrix {
+		namespaceName := string(entry.Metric["namespace"])
+		metricsByNamespace[namespaceName] = make(map[string][]model.SamplePair, 0)
+		metricsByNamespace[namespaceName]["memory"] = entry.Values
+		metricsByNamespace[namespaceName]["cpu"] = []model.SamplePair{}
+	}
+	for _, entry := range cpuMatrix {
+		namespaceName := string(entry.Metric["namespace"])
+		_, ok := metricsByNamespace[namespaceName]
+		if ok {
+			metricsByNamespace[namespaceName]["cpu"] = entry.Values
+		} else {
+			metricsByNamespace[namespaceName] = make(map[string][]model.SamplePair, 0)
+			metricsByNamespace[namespaceName]["cpu"] = entry.Values
+			metricsByNamespace[namespaceName]["memory"] = []model.SamplePair{}
+		}
+	}
+
+	// Construct output.
+	for namespaceName, readingsLists := range metricsByNamespace {
+		jointReadings, err := joinReadings(readingsLists["cpu"], readingsLists["memory"])
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to parse readings: ")
+		}
+
+		payload, err := json.MarshalIndent(jointReadings, "", "  ")
+		if err != nil {
+			return nil, errors.Wrap(err, "Error formatting readings: ")
+		}
+		path := []string{"utilisation", namespaceName}
+		output.SaveResult(c.BundlePath, filepath.Join(path...), bytes.NewBuffer(payload))
+	}
+	stopCh <- struct{}{}
+	return output, nil
+}
+
+func StartPortForwarding(
+	ctx context.Context,
+	restConfig *rest.Config,
+	out output.Output,
+	podName string,
+) (client prom.Client, stopCh chan struct{}, err error) {
+	var (
+		localPort int
+		lastErr   error
+	)
+
+	if err = wait.PollImmediateUntilWithContext(ctx, 1*time.Second, func(ctx context.Context) (bool, error) {
+		localPort, stopCh, lastErr = PortForward(
+			restConfig, "kommander", podName, 10902, out,
+		)
+		return lastErr == nil, nil
+	}); err != nil {
+		return nil, nil, fmt.Errorf("%v: %w", err, lastErr)
+	}
+
+	baseURL := fmt.Sprintf("http://localhost:%d", localPort)
+	prometheusClient, err := prom.NewClient(prom.Config{
+		Address: baseURL, RoundTripper: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create prometheus client: %w", err)
+	}
+
+	return prometheusClient, stopCh, err
+}
+
+type PortForwardFn = func(*rest.Config, string, string, int, output.Output) (int, chan struct{}, error)
+
+// PortForward creates an asynchronous port-forward to a pod on the cluster in order to
+// access it over the local network.
+func PortForward(cfg *rest.Config, ns, pod string, port int, out output.Output) (localPort int, stop chan struct{}, err error) {
```

Comment:
these port forwarding things shouldn't be part of this package. what do you think?